youtube_dl/extractor/twitch.py

```diff
@@ -77,15 +78,19 @@ def login_step(page, urlh, note, data):
             page_url = urlh.geturl()
             post_url = self._search_regex(
                 r'<form[^>]+action=(["\'])(?P<url>.+?)\1', page,
-                'post url', default=page_url, group='url')
+                'post url', default=self._LOGIN_POST_URL, group='url')
             post_url = urljoin(page_url, post_url)
 
-            headers = {'Referer': page_url}
+            headers = {
+                'Referer': page_url,
+                'Origin': page_url,
+                'Content-Type': 'text/plain;charset=UTF-8'
+            }
 
             try:
                 response = self._download_json(
                     post_url, None, note,
-                    data=urlencode_postdata(form),
+                    data=json.dumps(form),
```

Comment:
Incorrect.

Reply:
@dstftw could you point me to the preferred method, please? Sending `form` as is to the passport API results in a JSON error: `{"error_code":1002,"error_description":"failed to decode JSON"}`

Thanks!

Reply:
`data` must be `bytes` so you'll need to `encode` it.

Reply:
Do you mean with `json.dumps(form).encode()`?

Reply:
Yes.